pkg/manifest/manifest.go

```diff
@@ -0,0 +1,118 @@
+package manifest
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+// Manifest stores Kubernetes object in Raw from a file.
+// It stores the GroupVersionKind for the manifest.
+type Manifest struct {
+	// OriginalFilename is set to the filename this manifest was loaded from.
+	// It is not guaranteed to be set or be unique, but we will set it when
+	// loading from disk to provide better debuggability.
+	OriginalFilename string
+
+	Raw []byte
+	GVK schema.GroupVersionKind
```

Comment:
Obj gives the same information. Why a copy?

Reply:
It is currently accessed by CVO through both Obj and Raw.

Reply:
documented structure